### PR TITLE
[CI][Hardware][Intel GPU] add Intel GPU(XPU) ci pipeline

### PR DESCRIPTION
**Pull request by jikunshang**
#3814 add XPU ci in `test-template.j2` which is outdated and not be triggered.

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ vLLM is flexible and easy to use with:
 - Tensor parallelism support for distributed inference
 - Streaming outputs
 - OpenAI-compatible API server
-- Support NVIDIA GPUs, AMD GPUs, and Intel CPUs
+- Support NVIDIA GPUs, AMD GPUs, Intel CPUs and GPUs
```
Comment (WoosukKwon) [inline README.md:62]: Thanks for adding it. We will write a doc for hardware support in the near future :).

**Review by WoosukKwon — COMMENTED** [review]
@jikunshang Thanks for adding the CI
A quick question: IIRC, we will use the 8xMax 1100 machine for the CI/CD (basically the machine we used to use for our dev). If that's true,
1. Would this be enough for the tests? Could the queue be exploded as we have many PRs everyday?
2. Which machine can we use for our development?

**Comment by jikunshang** [comment]
> @jikunshang Thanks for adding the CI A quick question: IIRC, we will use the 8xMax 1100 machine for the CI/CD (basically the machine we used to use for our dev). If that's true,
> 
> 1. Would this be enough for the tests? Could the queue be exploded as we have many PRs everyday?
> 2. Which machine can we use for our development?

Yes, it's 8xMax 1100 machine. I noticed that each xpu ci run will take less than 5 mins, I think it's OK. We can add more agent to speed up if necessary.
You can use the node we provide earlier for development which is same machine for CI.

**Comment by WoosukKwon** [comment]
@jikunshang 

> We can add more agent to speed up if necessary.

Do you mean you can provide more GPU machine for CI if needed?

> You can use the node we provide earlier for development which is same machine for CI.

I believe we shouldn't use the same machine for CI and dev (at least they shouldn't share the same GPUs) because that can cause unexpected failures in CI.

**Comment by jikunshang** [comment]
> > We can add more agent to speed up if necessary.
> 
> Do you mean you can provide more GPU machine for CI if needed?

Sorry, we don't have more GPU machine available currently, I mean we can add more CI agents and one agent binding to one gpu device.


> > You can use the node we provide earlier for development which is same machine for CI.
> 
> I believe we shouldn't use the same machine for CI and dev (at least they shouldn't share the same GPUs) because that can cause unexpected failures in CI.

yes you are right, I think we can use device 4-7 as ci test cards and device 0-3 for development. 
you can specify target device by `export ONEAPI_DEVICE_SELECTOR=level_zero:0-3`
cc @zhouyuan

**Review by WoosukKwon — APPROVED** [review]
LGTM! Let's see if the resources are enough to cover our CI load and revert this back if the queue gets too long.

**Pull request merged**